src/Teams/Data/IRepository.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Teams.Data
+{
+    interface IRepository<T> where T : class
+    {
+        IQueryable<T> GetAll();
+        IQueryable<T> GetById(T Id);
```

Comment:
GetById means that we expecting single record that match unique identifier, no need to return IQueryable